aws/resource_aws_lambda_function.go

```diff
@@ -283,10 +284,18 @@ func updateComputedAttributesOnPublish(_ context.Context, d *schema.ResourceDiff
 		d.SetNewComputed("version")
 		d.SetNewComputed("qualified_arn")
 	}
+
 	return nil
 }
 
 func hasConfigChanges(d resourceDiffer) bool {
+	hasVpcConfigChange := d.HasChange("vpc_config")
+	if hasVpcConfigChange {
+		// This returns true when vpc_config is set even when the values are the same
+		o, n := d.GetChange("vpc_config")
+		hasVpcConfigChange = reflect.DeepEqual(o, n)
```

Comment:
It took me a while to see this, but this is setting `hasVpcConfigChange` to `true` when they're `DeepEqual`. However, since `vpc_config` contains `TypeSet` elements, it will always return not equal.

Reply:
I have tested this with a real Terraform project and this works: I don't have a plan with this change when a `lambda_function` has a `vpc_config`, and have a plan without that change. Additional debug logs show that `hasChange("vpc_config")` returns `true`, and `reflect.DeepEqual(o, n)` also returns true.

I have also tested it against a real change in `vpc_config` and made sure it shows up in plan.

However, since there can be only one `vpc_config`, if `hasChange("vpc_config.0.subnet_ids") || hasChange("vpc_config.0.security_group_ids")` does the trick, it seems to be a cleaner solution – it doesn't need reflection or conditionals. I'll test it.